openedx/openedx-platform
# Do not delete course enrollments

Ensure that course enrollments do not get deleted when the associated course overview gets recreated.

**JIRA tickets**: [OSPR-2535](https://openedx.atlassian.net/browse/OSPR-2535)

**Discussions**: https://github.com/edx/edx-platform/pull/18316#issuecomment-407264242

**Dependencies**: None

**Sandbox URL**:

* LMS: https://pr18657.sandbox.opencraft.hosting/
* Studio: https://studio-pr18657.sandbox.opencraft.hosting/

Ping us for shell access if required.

**Merge deadline**: ASAP -- needs to go into Hawthorn. CC @nedbat 

**Testing instructions**:

1. Locate an entry in the `course_overviews_courseoverview` where `version=4`, or modify an existing one to create this situation.
1. Ensure that the course has existing enrollments.
1. View the course in the LMS, both in the `/course` and `/instructor` pages.
1. Ensure that:
   * the `course_overviews_courseoverview` record is recreated with updated `version=6`.
   * the `student_courseenrollment` records count for the course remains unchanged.
   * no errors are shown in the GUI.

**Reviewers**
- [ ] OpenCraft reviewer TBD
- [ ] edX reviewer[s] TBD -- @doctoryes ?
